File: messanger.client/src/app/models/base/base-dto.ts
/** Базовый класс всех DTO. */
export class BaseDto {

  /** Идентификатор. */
  id: number;

  /** Сущность удалена? */
  isDeleted: boolean;

  /** Дата/время создания. */
  creationDateTime: Date;

  /** Дата/время последнего изменения. */
  lastModifyDateTime: Date;

  /** Идентификатор пользователя, последним измененившего сущность. */
  lastModifyUserId: number;

  constructor(
    id: number,
    isDeleted: boolean,
    creationDateTime: Date,
    lastModifyDateTime: Date,
    lastModifyUserId: number
  ) {
    this.id = id;
    this.isDeleted = isDeleted;
    this.creationDateTime = creationDateTime;
    this.lastModifyDateTime = lastModifyDateTime;
    this.lastModifyUserId = lastModifyUserId;
  }
}

File: messanger.Server/Core/Core.Dto/BaseDto.cs
namespace Core.Dto;

/// <summary>
/// Базовый класс всех dto
/// </summary>
public class BaseDto
{
    /// <summary>
    /// Идентификатор.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Сущность удалена?
    /// </summary>
    public bool IsDeleted { get; set; }

    /// <summary>
    /// Дата/время создания.
    /// </summary>
    public DateTime CreationDateTime { get; set; }

    /// <summary>
    /// Дата/время последнего изменения.
    /// </summary>
    public DateTime LastModifyDateTime { get; set; }

    /// <summary>
    /// Идентификатор пользователя, последним измененившего сущность.
    /// </summary>
    public int LastModifyUserId { get; set; }
}
